data/json/items/ammo/454.json

```diff
@@ -0,0 +1,25 @@
+[
```

Comment:
Presumably we could have the build process concatenate the `data/json` directory (possibly via `jq` script). Such a process could also sort the data in the optimal order considering the performance characteristics of the STL containers (subject to profiling). Finally that would also be a good time to start introducing validation of the JSON data at build time.

In that way we retain the readability advantage of the hierarchy but not the potential performance problems.

> I have an idea to concatenate all the json together and compress it to improve load times, that might mitigate/eliminate the issue.

How far on with this are you as I have some existing `jq` scripts I'm using and might be in a position to write a starting implementation?